python/pyspark/worker.py

```diff
@@ -177,8 +177,13 @@ def process():
             process()
     except Exception:
         try:
+            exc_info = traceback.format_exc()
+            try:
+                exc_info = exc_info.encode("utf-8")
```

Comment:
I think we should not just rely on the exception handling but only need to `encode` on `unicode` in Python 2 and `str` in Python 3.